Gifski/Utilities.swift

```diff
@@ -608,22 +598,87 @@ extension AVAsset {
 
 	This can be useful to trim blank frames from files produced by tools like the iOS simulator screen recorder.
 	*/
-	func trimBlankFrames() throws -> AVAsset {
+	func trimmingBlankFrames() throws -> AVAsset {
 		guard let videoTrack = self.firstVideoTrack else {
-			throw VideoTrimmingError.assetIsMissingTrack
+			throw AVAssetTrack.VideoTrimmingError.assetIsMissingTrack
 		}
 
-		let trimmedTrack = try videoTrack.trimBlankFrames()
+		let trimmedTrack = try videoTrack.trimmingBlankFrames()
 
 		guard let trimmedAsset = trimmedTrack.asset else {
-			throw VideoTrimmingError.assetIsNil
+			throw AVAssetTrack.VideoTrimmingError.assetIsNil
 		}
 
 		return trimmedAsset
 	}
 }
 
 
+extension AVAssetTrack {
+	enum VideoTrimmingError: String, Error {
+		case assetIsNil
+		case assetReaderFailure
+		case assetIsEmpty
+		case assetIsMissingTrack
+		case compositionCouldNotBeCreated
+		case compositionIsMissingTrack
+	}
+
+
+	/**
+	Removes blank frames from the begining of the track.
+
+	This can be useful to trim blank frames from files produced by tools like the iOS simulator screen recorder.
+	*/
+	func trimmingBlankFrames() throws -> AVAssetTrack {
+		// Create new composition
+		let composition = AVMutableComposition()
+		guard
+			let wrappedTrack = composition.addMutableTrack(withMediaType: mediaType, preferredTrackID: .zero)
+		else {
+			throw VideoTrimmingError.compositionCouldNotBeCreated
+		}
+		try wrappedTrack.insertTimeRange(timeRange, of: self, at: .zero)
+
+		let reader = try AVAssetReader(asset: composition)
+
+		// Create reader for wrapped track
+		let readerOutput = AVAssetReaderTrackOutput(track: wrappedTrack, outputSettings: nil)
+		reader.add(readerOutput)
+		reader.startReading()
+
+		defer {
+			reader.cancelReading()
+		}
+
+		// Iterate through samples until we reach one with a non-zero size
+		while let sampleBuffer = readerOutput.copyNextSampleBuffer() {
+			guard [.completed, .reading].contains(reader.status) else {
+				throw reader.error ?? VideoTrimmingError.assetReaderFailure
+			}
+
+			// On first non-empty frame
+			guard sampleBuffer.totalSampleSize == 0 else {
+				let currentTimestamp = sampleBuffer.outputPresentationTimeStamp
+
+				wrappedTrack.removeTimeRange(.init(start: .zero, end: currentTimestamp))
+
+				return wrappedTrack
+			}
+		}
+
+		throw VideoTrimmingError.assetIsEmpty
+	}
+}
+
+
+extension AVAssetTrack.VideoTrimmingError: LocalizedError {
+	public var errorDescription: String? {
+		self.rawValue
+	}
+}
```

Comment:
This is an abuse of `LocalizedError`. You need to actually do a switch where you write human-readable messages for each error.

Reply:
Fixed in 3a2c8a6.